### DISCUSSION TITLE
v0.31.1

### DISCUSSION THREAD

**Discussion by rfratto** [grafana-cold-storage/agent#2933] [Announcements]
This is release `v0.31.1` of the Grafana Agent.

### Upgrading
Read the [upgrade guide](https://grafana.com/docs/agent/v0.31/upgrade-guide) for specific instructions on upgrading from older versions.

#### Breaking changes

- All release Windows `.exe` files are now published as a zip archive.
  Previously, `grafana-agent-installer.exe` was unzipped. (@rfratto)

#### Other changes

- Support Go 1.20 for builds. (@rfratto)

### Installation:
Grafana Agent is currently distributed in plain binary form, Docker container images, a Windows installer, and a Kubernetes install script. Choose whichever fits your use-case best.

#### Kubernetes

Install directions [here.](https://grafana.com/docs/grafana-cloud/quickstart/agent-k8s/)

#### Docker container:

* https://hub.docker.com/r/grafana/agent

Docker containers are published as `grafana/agent:v0.31.1`. For Windows Docker containers, use `grafana/agent:v0.31.1-windows` instead. 

#### Windows installer

The Windows installer is provided as a [release asset](https://github.com/grafana/agent/releases/download/v0.31.1/grafana-agent-installer.exe.zip) for x64 machines.

#### Binary

We provide precompiled binary executables for the most common operating systems. Choose from the assets below for your matching operating system.

Note: ppc64le builds are currently considered secondary release targets and do not have the same level of support and testing as other platforms.

Example for the `linux` operating system on `amd64`:

```bash
# download the binary
curl -O -L "https://github.com/grafana/agent/releases/download/v0.31.1/grafana-agent-linux-amd64.zip"

# extract the binary
unzip "grafana-agent-linux-amd64.zip"

# make sure it is executable
chmod a+x "grafana-agent-linux-amd64"
```

#### `agentctl`

`agentctl`, a tool for helping you interact with the Agent, is available as a Docker image:

Docker containers are published as `grafana/agentctl:v0.31.1`. For Windows Docker containers, use `grafana/agentctl:v0.31.1-windows` instead. 

Or as a binary. Like before, choose the assets below that matches your operating system. For example, with `linux` on `amd64`:

```bash
# download the binary
curl -O -L "https://github.com/grafana/agent/releases/download/v0.31.1/grafana-agentctl-linux-amd64.zip"

# extract the binary
unzip "grafana-agentctl-linux-amd64.zip"

# make sure it is executable
chmod a+x "grafana-agentctl-linux-amd64"
```

#### `agent-operator`

`agent-operator`, a Kubernetes Operator for the Grafana Agent, is available only as a Docker image:

```bash
docker pull "grafana/agent-operator:v0.31.1"
```

<hr /><em>This discussion was created from the release <a href='https://github.com/grafana/agent/releases/tag/v0.31.1'>v0.31.1</a>.</em>

**Comment by Fydon** [comment]
@rfratto Thank you for this. Now that the Windows `.exe` files are all zipped, the release asset link for the installer in the notes does not work.

I assume that the zips are due to some browsers flagging them and Windows blocking them?

**Reply by rfratto** [reply]
Thanks for pointing that out, I'll fix the instructions :) 

> I assume that the zips are due to some browsers flagging them and Windows blocking them?
 
It's mainly for compression and being consistent with the other zips.
